.github/workflows/cli-ci.yml

```diff
@@ -0,0 +1,89 @@
+name: cli/ubi CI
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'cli/go.mod'
+      - 'cli/ubi.go'
+      - 'bin/ubi'
+      - 'spec/cli_spec.rb'
+  pull_request:
+    paths:
+      - 'cli/go.mod'
+      - 'cli/ubi.go'
+      - 'bin/ubi'
+      - 'spec/cli_spec.rb'
+
+jobs:
+  cli-ci:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubicloud, ubicloud-arm]
+    name: cli/ubi CI - ${{matrix.runs-on}}
+    runs-on: ${{matrix.runs-on}}
+
+    env:
+      DB_USER: clover
+      DB_PASSWORD: nonempty
+      DB_NAME: clover_test
+
+    services:
+      postgres:
+        image: postgres:15.4
+        env:
+          POSTGRES_USER: ${{ env.DB_USER }}
+          POSTGRES_PASSWORD: ${{ env.DB_PASSWORD }}
+          POSTGRES_DB: ${{ env.DB_NAME }}
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+    - name: Perform superuser-only actions, then remove superuser
+      run: |
+        psql "postgres://${{ env.DB_USER }}:${{ env.DB_PASSWORD }}@localhost:5432/${{ env.DB_NAME }}" \
+          -c "CREATE EXTENSION citext; CREATE EXTENSION btree_gist; CREATE ROLE clover_password PASSWORD '${{ env.DB_PASSWORD }}' LOGIN; ALTER ROLE ${{ env.DB_USER }} NOSUPERUSER"
+
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Install ruby for ARM runners if not cached
+      if: matrix.runs-on == 'ubicloud-arm'
+      run: |
+        RUBY_VERSION="$(grep -E '^ruby ' .tool-versions | cut -d' ' -f2)"
+        if [ ! -f $RUNNER_TOOL_CACHE/Ruby/$RUBY_VERSION/arm64.complete ]; then
+          git clone https://github.com/rbenv/ruby-build.git
+          sudo ./ruby-build/install.sh
+          sudo ruby-build "$RUBY_VERSION" $RUNNER_TOOL_CACHE/Ruby/$RUBY_VERSION/arm64
+          sudo chown -R runner:runner $RUNNER_TOOL_CACHE/Ruby
+          touch $RUNNER_TOOL_CACHE/Ruby/$RUBY_VERSION/arm64.complete
+          rm -rf ruby-build
+        fi
+
+    - name: Set up Ruby
```

Comment:
Similar to this, you will probably want to utilize the `setup-go` action so you can ensure the expected go version is being used, rather than whatever happens to be on the image.

Reply:
I'm guessing this works now because I'm using the distro-provided go package, and the runner is running the same OS.  I'll work on this.

Reply:
Updated cli-ci.yml to use setup-go with a matching version.